packages/cloud_firestore/lib/src/blob.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright 2018, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of cloud_firestore;
+
+class Blob {
+  final Uint8List bytes;
+  const Blob(this.bytes);
+
+  @override
+  bool operator ==(dynamic other) =>
+      other is Blob && other.hashCode == hashCode;
```

Comment:
Two blobs are equal if they contain the same sequence of bytes. Hash codes may collide.